ts/WoltLabSuite/Core/Ui/ItemList/LineBreakSeparatedText.ts

```diff
@@ -0,0 +1,258 @@
+    inputAddon.appendChild(addButton);
+
+    this.clearButton = document.createElement("a");
+    this.clearButton.href = "#";
+    this.clearButton.classList.add("button", "inputSuffix", "jsTooltip");
+    this.clearButton.title = Language.get("wcf.global.button.delete");
+    this.clearButton.innerHTML = '<span class="icon icon16 fa-times"></span>';
+    this.clearButton.addEventListener("click", (ev) => this.clearList(ev));
+    inputAddon.appendChild(this.clearButton);
+    if (this.items.size === 0) {
+      DomUtil.hide(this.clearButton);
+    }
+  }
+
+  /**
+   * Clears the item list after clicking on the clear button.
+   */
+  protected clearList(ev: Event): void {
+    ev.preventDefault();
+
+    UiConfirmation.show({
+      confirm: () => {
+        this.itemList.innerHTML = "";
+        this.items.clear();
+
+        this.hideList();
+      },
+      message: Language.get("wcf.acp.option.type.lineBreakSeparatedText.clearList.confirmMessage"),
+      messageIsHtml: true,
+    });
+  }
+
+  /**
+   * Deletes an item from the list after clicking on its delete icon.
+   */
+  protected deleteItem(event: Event): void {
+    const button = event.currentTarget as HTMLElement;
+    const item = button.closest("li")!.dataset.value!;
+
+    UiConfirmation.show({
+      confirm: () => {
+        button.closest("li")!.remove();
+
+        if (this.itemList.childElementCount === 0) {
+          this.hideList();
+        }
+
+        this.items.delete(item);
+      },
+      message: Language.get("wcf.button.delete.confirmMessage", {
+        objectTitle: item,
+      }),
+      messageIsHtml: true,
+    });
+  }
+
+  /**
+   * Hides the item list and clear button.
+   */
+  protected hideList(): void {
+    DomUtil.hide(this.itemList);
+    DomUtil.hide(this.clearButton!);
+  }
+
+  /**
+   * Adds the initial values to the list.
+   */
+  protected initValues(): void {
+    Array.from(this.itemList.children).forEach((el: HTMLElement) => {
+      this.items.add(el.dataset.value!);
+
+      el.querySelector(".jsDeleteItem")!.addEventListener("click", (ev) => this.deleteItem(ev));
+    });
+  }
+
+  /**
+   * Inserts the given item to the list.
+   */
+  protected insertItem(item: string): void {
+    this.items.add(item);
+
+    const itemElement = document.createElement("li");
+    itemElement.dataset.value = item;
+
+    const deleteButton = document.createElement("span");
+    deleteButton.classList.add("icon", "icon16", "fa-times", "jsDeleteItem", "jsTooltip", "pointer");
+    deleteButton.title = Language.get("wcf.global.button.delete");
+    deleteButton.addEventListener("click", (ev) => this.deleteItem(ev));
+    itemElement.append(deleteButton);
+
+    itemElement.append(document.createTextNode(" "));
+
+    const label = document.createElement("span");
+    label.innerText = item;
+    itemElement.append(label);
+
+    Array.from(this.itemList.children).some((el: HTMLElement) => {
```

Comment:
Do not use `.some()` while ignoring the return value. While the behavior is well-defined it is very non-obvious usage. I believe you can do `const next = yadayada.find((el) => el.dataset.value!.localeCompare(item) === 1)` here.